### DISCUSSION TITLE
Getting all the routes of denmark or copenhagen just like how it is in the map

### DISCUSSION THREAD

**Discussion by jayeshdesai4520** [transitland/transitland#399] [Q&A] [answered]
Hey everyone,
I am trying to fetch all the routes of Copenhagen, just like it is shown on the transit.land map,
so what I am doing is, I am creating a b-box with all lon and lat info of Copenhagen and display it on map, using v1 API

`import folium
from IPython.display import display
import requests
from shapely.geometry import LineString

api_key = "" 
copenhagen_coordinates = (55.6761, 12.5683)  # Latitude, Longitude of Copenhagen
 
padding = 0.01
min_lat = copenhagen_coordinates[0] - padding
max_lat = copenhagen_coordinates[0] + padding
min_lon = copenhagen_coordinates[1] - padding
max_lon = copenhagen_coordinates[1] + padding

url = f"https://transit.land/api/v1/routes?bbox={min_lon},{min_lat},{max_lon},{max_lat}&per_page=1000&api_key={api_key}"
response = requests.get(url)

 if response.status_code == 200:
    routes_data = response.json()
    map_center = copenhagen_coordinates
    map = folium.Map(location=map_center, zoom_start=12)

    for route in routes_data['routes']:
        route_name = route['name']
        route_geometry = route['geometry']
        if isinstance(route_geometry, dict) and 'coordinates' in route_geometry:
            coordinates = route_geometry['coordinates']
            for coords in coordinates:
                line = LineString(coords)
                latitudes, longitudes = zip(*coords)
                folium.PolyLine(locations=list(zip(longitudes, latitudes)), color='blue', weight=3, popup=route_name).add_to(map)

    # Display the map
     display(map)
 else:
     print("Error: Failed to retrieve transit routes.")
`
So this code only fetches one route which is flixbus, its not able to fetch all transit routes, it works well on Paris and Toronto coordinates!

**Comment by drewda** [comment, marked as answer]
The v1 Datastore API is deprecated. The v1 endpoints are no longer able to return data for all feeds.

Please use the v2 REST API. For more information, see https://www.transit.land/documentation/rest-api/routes

Or to display all routes, you can also use the v2 Vector Tiles: https://www.transit.land/documentation/vector-tiles

**Reply by jayesh0405** [reply]
Thanks for the answer,

But I don't see BBOX in version 2. Is there any alternative?

Thanks,
Jayesh Desai

**Reply by drewda** [reply]
You're right that there is no bbox query parameter in the v2 REST stops endpoint. It was a very expensive query to run.

You can use an alternative strategy such as querying other endpoints for operators or feeds for a given location, using one or more of those Onestop IDs, and filtering by either `feed_onestop_id` or `operator_onestop_id` in the routes endpoint.
